Jenkinsfile

```diff
@@ -16,22 +16,29 @@ node('openshift-build-1') {
     timeout(time: 30, unit: 'MINUTES') {
       deleteDir()
       set_workspace()
-      stage('clone') {
-        dir('aos-cd-jobs') {
+      dir('aos-cd-jobs') {
+        stage('clone') {
           checkout scm
           sh 'git checkout master'
         }
-      }
-      stage('run') {
-        sshagent(['openshift-bot']) { // git repo privileges stored in Jenkins credential store
-          sh '''\
-virtualenv env/
-. env/bin/activate
+        stage('run') {
+          final url = sh(
+            returnStdout: true,
+            script: 'git config remote.origin.url')
+          def prune = 'true', key = 'openshift-bot'
+          if(url.trim() != 'git@github.com:openshift/aos-cd-jobs.git') {
+            prune = ''
+            key = "${(url =~ /.*:([^\/]+)/)[0][1]}-aos-cd-bot"
```

Comment:
Can we add a check at the top of this job.. if (url startsWith "git" == false ) then error(please use an ssh url)?
I know the job won't work without it, but a clean error will save time diagnosing.

Reply:
Changed, but I could not find a way to extract the git remote url from jenkins,
so the check has to be done after cloning.